### PR TITLE
Basic CEC support. 

### LINKED ISSUE #358: Feature Request: basic HDMI-CEC support.
Pulse-Eight sells a USB HDMI CEC adapter. It is the same one used in the raspberry pi as they provide the module for it. The adapter is cheap and can be used to do things like switch the TV input.

The following library includes a very simple to use command line client: https://github.com/Pulse-Eight/libcec. I believe an arch package exists for it.

I think it would greatly enhance the console experience if a command to turn the input active during bootup and a command to disable the input during shutdown was implemented. Additionally, turning the input off for suspend when pressing the power button and turning it on in resume would be desirable. This is similar to how Xbox implements their CEC so that you can turn a controller on and it wakes the console and turns the TV on. Waking the system with a USB Xbox wireless adapter already works, so cec-client is the only missing piece.

Turning the display off and on should be as simple as a systemd service that calls `echo as | cec-client -s` to turn it on and `echo standby 0 | cec-client -s` to turn it off during boot and shutdown respectively. If no CEC dongle is available this just won't do anything and is safe.

### PR DESCRIPTION
**Pull request by BoukeHaarsma23**
This PR adds basic switching on the TV when booting up, and powering off when shutting down. This is safe to use even without the CEC adapter.

I'm unsure whether or not we should make these configurable, but I would like to know how you guys think about this.

Closes #358.

### CODE REVIEW

File: rootfs/etc/systemd/system/cec-active-source.timer [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Unit]
```
Comment (Samsagax) [inline rootfs/etc/systemd/system/cec-active-source.timer:1]: Is this file needed at all? Would it be best to add WantedBy and Before/After lines in the service itself? This seems like a dirty hack.

File: rootfs/usr/bin/cec-control [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+ACTION="$1"
+
+
+# Source environment from ~/.config/environment.d with user overrides
+set -a
+for i in /home/gamer/.config/environment.d/*.conf ;
```
[the remark this answers is not in the document]
Comment (Samsagax) [reply rootfs/usr/bin/cec-control:7]: Does the service need to be run as root? If not, it can be a user service and you' ll know for sure the $HOME variable. If not, you can have a configuration file under `/etc` to configure the daemon.

File: rootfs/etc/systemd/system/cec-poweroff-tv.service [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+Description=Use CEC to power off TV
+DefaultDependencies=no
+Before=final.target
+
+[Service]
+Type=oneshot
+ExecStart=bash -c "echo 'standby 0' | cec-client -s"
+
+[Install]
+WantedBy=poweroff.target
```
[the remark this answers is not in the document]
Comment (MiningMarsh) [reply rootfs/etc/systemd/system/cec-poweroff-tv.service]: Standby 0 does not power off the TV, it marks the input as in standby. The TV (and mine does do this, tested using Kodi which has libCEC) will just switch to another active input and only suspends if evetr input is on standby.

I.e., it you have your Xbox on screen and suddenly your chimeraOS box issues a standby, the TV won't turn off, it will just internally mark that the chimeraOS input is in standby.

**Comment by MiningMarsh** [comment]
I have am attached HDMI-CEC adapter already on my machine. After code review I'd be willing to test builds containing this new functionality if desired and convenient to the chimera developers.

File: rootfs/usr/bin/cec-control (hunk 2 above)
[the remark this answers is not in the document]
Comment (pastaq) [reply rootfs/usr/bin/cec-control:7]: Why does the user need to be added to the dialout group? Cold a UDEV rule for the CEC device give the user full permissions without changing the group permissions?

[the remark this answers is not in the document]
Comment (pastaq) [reply rootfs/usr/bin/cec-control:7]: Any traction on this? IMO running as a user service seems like the best course, I'm against using hard coded directories for user when environment variables are set.  I'm waiting to test this on my CEC controller once these changes are made.

**Review by pastaq — CHANGES_REQUESTED** [review]

**Review by pastaq — CHANGES_REQUESTED** [review]
I think we should go with a user service and use environment variables. Adding gamer to the dial out group is trivial and I see no reason root should be managing this. Standing by to test on my adapter once these changes are ready.

File: rootfs/usr/bin/cec-control [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+do
+    [[ -f "${i}" ]] && . "${i}"
+done
+set +a
+
+# OVERRIDE WITH DEFAULTS IF NOT SET
```
Comment (pastaq) [inline rootfs/usr/bin/cec-control:13]: If a config file isn't present for this we should generate a default one and then read from it, so the user can identify how to change these behaviors. Hiding this in code without documentation or example would make using these as environment variables pointless.

**Review by pastaq — CHANGES_REQUESTED** [review]

**Review by Samsagax — CHANGES_REQUESTED** [review]
Systemd units should be added under `/usr/lib` instead of `/etc`.
I'm not convinced on the configuration part. This should be a daemon or a root service. In that case configuration should be done by a conf file under `/etc` or something similar as it is per machine.
